### DISCUSSION TITLE
Triggerer getting restarted multiple times

### DISCUSSION THREAD

**Discussion by EMT-Swaroop** [apache/airflow#41945] [General]
Hey Team,

We are using Airflow 2.9.3 version in dev, preprod and prod. But in preprod and prod triggerer is continuously getting restarted but not in dev. We have checked the resource limits, and the consumption is normal for CPU and memory. No DAG changes have also been done. Nothing from k8s side as well.

Below are the logs we are getting

[Info] Handling Signal: term
[Info] Worker exiting (pid: 24)
[Info] Worker exiting (pid: 25)
{triggerer_job_runner.py: 350} Exited Trigger Loop

We are using the default probe values from parent chart.

Please let me know if there is 
1. a fix 
2. we can resolve it by doing configuration change 
3. there is any related git hub issue or discussion or related article
4. someone facing the same issue for "triggerer"

Thanks in advance

**Comment by potiuk** [comment]
Look at your liveness probes. You need to find out what is sending the TERM signal - most likely liveness probe and in any case your kubernetes deployment logs should tell you why your pods are getting terminated - there are many reasons and guessing it is much less effective than looking in the logs of k8s. 

Which is something you need to be able to do for any app running on k8s - this is not specific to Airflow. And there is no-one else who can do it by you (or whoever manages k8s of yours)./

**Reply by potiuk** [reply]
And you will find a lot of "how tos" when you search for liveness probes and looking at your k8s logs in general.
